format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/SplitQuotesIterator.java

```diff
@@ -31,14 +34,20 @@
  */
 public class SplitQuotesIterator extends AbstractIterator<String> {
   private static final char QUOTE_CHAR = '\"';
-  private final String delimitedString;
+  private static final String LINE_SEPARATOR = "\n";
   private final String delimiter;
+  private final RecordReader recordReader;
+  private final boolean enableMultilineSupport;
+  private String delimitedString;
   private int index;
   private boolean endingWithDelimiter = false;
 
-  public SplitQuotesIterator(String delimitedString, String delimiter) {
+  public SplitQuotesIterator(String delimitedString, String delimiter, RecordReader recordReader,
```

Comment:
need to annotate recordReader as `@Nullable` and handle the case when it's null (it's null when used in DataTypeDetectorUtils)

Reply:
Annotated. and it is null when enableMultiLine support flag is false. and this will used only when flag is true